ue4docker/infrastructure/WindowsUtils.py

```diff
@@ -10,18 +10,20 @@ class WindowsUtils(object):
 	# The oldest Windows build we support
 	_minimumRequiredBuild = 17763
 
-	# The latest Windows build version we recognise as a non-Insider build
-	_latestReleaseBuild = 19042
+	# This lookup table is based on the list of valid tags from <https://hub.docker.com/r/microsoft/windowsservercore/>
+	# and list of build-to-release mapping from https://docs.microsoft.com/en-us/windows/release-health/release-information
+	_tagsByBuildNumber = {
+		17763: 'ltsc2019',
+		19041: '2004',
+		19042: '20H2',
+	}
 
 	# The list of Windows Server Core base image tags that we recognise, in ascending version number order
-	_validTags = ['ltsc2019', '1903', '1909', '2004', '20H2']
+	_validTags = list(_tagsByBuildNumber.values())
 
 	# The list of Windows Server and Windows 10 host OS releases that are blacklisted due to critical bugs
 	# (See: <https://unrealcontainers.com/docs/concepts/windows-containers>)
-	_blacklistedReleases = ['1903', '1909']
-
-	# The list of Windows Server Core container image releases that are unsupported due to having reached EOL
-	_eolReleases = ['1903', '1909']
```

Comment:
Windows 10 2004 hits EOL in December, and 20H2 EOLs in May next year. So if we just disappear them entirely, then we lose support for any Windows Client platforms.

I'd keep the EOL support around until we either decide to drop Windows 10 process isolation support once there's no supported base images available, or something changes on the MS side, or some other intervening idea occurs.

Better to say "Yes, we recognise this, it's EOL, are you sure you want to continue" when we can recognise a version.

Also, these two versions are only blacklisted for Docker versions before 19.03.6, so this change actually _promotes_ them from EOL warning with user-override to "supported", i.e. silently accepted.

Reply:
The trick is that they're blacklisted as a _host_ and EOL as a _target_. We silently allow building supported versions on EOL hosts.

Reply:
> Windows 10 2004 hits EOL in December, and 20H2 EOLs in May next year. So if we just disappear them entirely, then we lose support for any Windows Client platforms.

OK, agreed.

Reply:
Ah, good point. I'm fine with not complaining about using an EOL host to build a supported target.

I'm ambivalent about complaining when we try to use an EOL _target_; I'd lean towards keeping the warning around based on the calculated base tag (i.e. _after_ applying the user's specified base tag), but that isn't really comprehensive or reliable (e.g., basetag `1909-KB4574727-amd64` would not be recognised), so I don't mind if we drop it now.

My personal use-cases are 100% process-isolation-based (and Windows Server LTSC-based 95% of the time), so I haven't had any value from the EOL warning myself.

I'd be interested to know if @adamrehn has any concerns about dropping this (or use-cases where it prevents a bad outcome), but if not, then I'm happy to resolve this thread.

Reply:
> basetag 1909-KB4574727-amd64 would not be recognised

ue4-docker doesn't support building targets that are not hardcoded in valid tags list _at all_. #166 wanted to change that, but kinda failed.

----

I thought about 20H2 again. I _very_ doubt that anyone would want to build ue4-docker images for it after it gets EOLed. If they're on a newer host, they would need to use Hyper-V isolation. And if you use Hyper-V isolation, why not just use a supported target version? Alternatively, they still run EOLed 20H2 on their host. Well, if you use EOLed Windows, also use EOLed version of ue4-docker that has 20H2 support, no?